src/machines/java/com/enderio/machines/common/blockentity/sync/LargeMachineEnergyDataSlot.java

```diff
@@ -37,13 +38,31 @@ protected IMachineEnergyStorage valueFromNBT(Tag nbt) {
             long maxStored = compoundTag.getLong(MachineNBTKeys.ENERGY_MAX_STORED);
             return new LargeImmutableMachineEnergyStorage(energy, maxStored);
         } else {
-            throw new IllegalStateException("Invalid compound tag was passed over the network.");
+            throw new IllegalStateException("Invalid LargeMachineEnergy was passed over the network.");
+        }
+    }
+
+    @Override
+    public void toBuffer(FriendlyByteBuf buf, IMachineEnergyStorage value) {
+        ILargeMachineEnergyStorage storage = (ILargeMachineEnergyStorage) getter.get(); //why the getter?
+        buf.writeLong(storage.getLargeEnergyStored());
+        buf.writeLong(storage.getLargeMaxEnergyStored());
+    }
+
+    @Override
+    public IMachineEnergyStorage valueFromBuffer(FriendlyByteBuf buf) {
+        try {
+            long energy = buf.readLong();
+            long maxStored = buf.readLong();
+            return new LargeImmutableMachineEnergyStorage(energy, maxStored);
+        } catch (Exception e) {
+            throw new IllegalStateException("Invalid LargeMachineEnergy buffer was passed over the network.");
         }
     }
 
     @Override
     protected int hashCode(IMachineEnergyStorage value) {
-        var largeStorage = (ILargeMachineEnergyStorage)value;
+        var largeStorage = (ILargeMachineEnergyStorage)value; //why the cast?
```

Comment:
We need access to the getLarge... methods in this interface

Reply:
yes but why is this not a "Large" slot? We don't actually use the IMachineEnergyStorage it seems?

Reply:
Hmm good point, could change the generic type and see if it works

Reply:
So this might not have worked, as right with this change, the banks are limited to an int.

Reply:
This also happens in the current branch, so it is unrelated to the PR.